pyomo/contrib/fme/plugins/fourier_motzkin_elimination.py

```diff
@@ -0,0 +1,329 @@
+                # is a variable we are projecting out
+                if obj not in vars_to_eliminate:
+                    continue
+                if obj.lb is not None:
+                    constraints.append({'body': generate_standard_repn(obj),
+                                        'upper': None,
+                                        'lower': value(obj.lb),
+                                        'map': ComponentMap([(obj, 1)])})
+                if obj.ub is not None:
+                    constraints.append({'body': generate_standard_repn(-obj),
+                                        'upper': None,
+                                        'lower': -value(obj.ub),
+                                        'map': ComponentMap([(obj, -1)])})
+            else:
+                raise RuntimeError(
+                    "Found active component %s of type %s. The "
+                    "Fourier-Motzkin Elimination transformation can only "
+                    "handle purely algebraic models. That is, only "
+                    "Sets, Params, Vars, Constraints, Expressions, Blocks, "
+                    "and Objectives may be active on the model." % (obj.name, 
+                                                                    obj.type()))
+
+        new_constraints = self._fourier_motzkin_elimination(constraints,
+                                                            vars_to_eliminate)
+
+        # put the new constraints on the transformation block
+        for cons in new_constraints:
+            body = cons['body']
+            lhs = sum(coef*var for (coef, var) in zip(body.linear_coefs,
+                                                      body.linear_vars))
+            lower = cons['lower']
+            if type(lhs >= lower) is bool:
+                if lhs >= lower:
+                    continue
+                else:
+                    # This would actually make a lot of sense in this case...
+                    #projected_constraints.add(Constraint.Infeasible)
+                    raise RuntimeError("Fourier-Motzkin found that model is "
+                                       "infeasible!")
+            else:
+                projected_constraints.add(lhs >= lower)
+
+    def _process_constraint(self, constraint):
+        """Transforms a pyomo Constraint objective into a list of dictionaries
+        representing only >= constraints. That is, if the constraint has both an
+        ub and a lb, it is transformed into two constraints. Otherwise it is
+        flipped if it is <=. Each dictionary contains the keys 'lower', 'upper'
+        and 'body' where, after the process, 'upper' will be None, 'lower' will
+        be a constant, and 'body' will be the standard repn of the body.
+        (The constant will be moved to the RHS).
+        """
+        body = constraint.body
+        std_repn = generate_standard_repn(body)
+        # linear only!!
+        if not std_repn.is_linear():
```

Comment:
It looks like it does the right thing. Assuming you mean something like:
```
m = ConcreteModel()
m.p = Param(initialize=5, mutable=True)
m.p2 = Param(initialize=5)
m.x = Var()
m.x.fix(3)
# All these are True:
generate_standard_repn(m.p).is_linear() 
generate_standard_repn(m.p2).is_linear() 
generate_standard_repn(5).is_linear() 
generate_standard_repn(m.x).is_linear() 
```
But these constraints can't possibly be needed since they have no variables, so they'll get filtered out after they make it through that check.